xds/src/main/java/io/grpc/xds/FallbackLb.java

```diff
@@ -58,55 +45,18 @@ protected LoadBalancer delegate() {
     return fallbackPolicyLb;
   }
 
-  @SuppressWarnings("deprecation")
   @Override
   public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
-    Attributes attributes = resolvedAddresses.getAttributes();
-    XdsConfig xdsConfig;
-    Object lbConfig = resolvedAddresses.getLoadBalancingPolicyConfig();
-    if (lbConfig != null) {
-      if (!(lbConfig instanceof XdsConfig)) {
-        fallbackLbHelper.updateBalancingState(
-            TRANSIENT_FAILURE,
-            new ErrorPicker(Status.UNAVAILABLE.withDescription(
-                "Load balancing config '" + lbConfig + "' is not an XdsConfig")));
-        return;
-      }
-      xdsConfig = (XdsConfig) lbConfig;
-    } else {
-      // In the future, in all cases xdsConfig can be obtained directly by
-      // resolvedAddresses.getLoadBalancingPolicyConfig().
-      Map<String, ?> newRawLbConfig = attributes.get(ATTR_LOAD_BALANCING_CONFIG);
-      if (newRawLbConfig == null) {
-        // This will not happen when the service config error handling is implemented.
-        // For now simply go to TRANSIENT_FAILURE.
-        fallbackLbHelper.updateBalancingState(
-            TRANSIENT_FAILURE,
-            new ErrorPicker(
-                Status.UNAVAILABLE.withDescription("ATTR_LOAD_BALANCING_CONFIG not available")));
-        return;
-      }
-      ConfigOrError cfg =
-          XdsLoadBalancerProvider.parseLoadBalancingConfigPolicy(newRawLbConfig, lbRegistry);
-      if (cfg.getError() != null) {
-        // This will not happen when the service config error handling is implemented.
-        // For now simply go to TRANSIENT_FAILURE.
-        fallbackLbHelper.updateBalancingState(TRANSIENT_FAILURE, new ErrorPicker(cfg.getError()));
-        return;
-      }
-      xdsConfig = (XdsConfig) cfg.getConfig();
-    }
-
-    LbConfig fallbackPolicy = xdsConfig.fallbackPolicy;
+    XdsConfig xdsConfig = (XdsConfig) resolvedAddresses.getLoadBalancingPolicyConfig();
```

Comment:
I am thinking that, instead of receiving the entire `XdsConfig`, would it be better to just received the parsed fallback policy (i.e., `PolicySelection`). In this way, `FallbackLb` does not receive anything extra other than the policy it needs to instantiate.

Reply:
That's true, however FallbackLb is not actually used and eventually will be re-designed, I don't want to touch this class too much in this PR..